modules/imap/functions.php

```diff
@@ -1093,7 +1093,7 @@ function clear_existing_reply_details($session) {
  */
 if (!hm_exists('imap_authed')) {
 function imap_authed($imap) {
-    return is_object($imap) && ($imap->get_state() == 'authenticated' || $imap->get_state() == 'selected');
+    return is_object($imap) && ($imap->get_state() == 'authenticated' || $imap->get_state() == 'selected' || $imap->get_state() == 'connected');
```

Comment:
If oauth2 returns connected instead of authenticated, let's handle that in the imap get_state() method. We have this refactored in the upcoming https://github.com/cypht-org/cypht/pull/1278, so we need a reliable way to check if imap server connection is ready to use or not. Instead of scattering this check throughout the modules, let's check it only in one place.